agent/src/main/java/reactor/blockhound/BlockHound.java

```diff
@@ -403,6 +403,8 @@ public void install() {
             // Eagerly trigger the classloading of `dynamicThreadPredicate` (since classloading is blocking)
             dynamicThreadPredicate.test(Thread.currentThread());
             BlockHoundRuntime.dynamicThreadPredicate = dynamicThreadPredicate;
+            // Null out the state so it can be evaluated with the configured dynamicThreadPredicate
+            BlockHoundRuntime.STATE.set(null);
```

Comment:
FYI it should be `ThreadLocal#remove` as seen here:
https://github.com/reactor/BlockHound/pull/124/files#diff-21558a210d558456b0f5da4268364dc3R414